src/main/java/org/sonatype/repository/conan/internal/proxy/ConanProxyRecipe.groovy

```diff
@@ -81,22 +70,26 @@ class ConanProxyRecipe
 
   @Override
   void apply(@Nonnull final Repository repository) throws Exception {
+    def conanProxyFacet = proxyFacet.get()
+    conanProxyFacet.configureDynamicMatcher(viewClosure)
+
     repository.attach(securityFacet.get())
-    repository.attach(configure(viewFacet.get()))
+    repository.attach(viewFacet.get())
     repository.attach(httpClientFacet.get())
     repository.attach(negativeCacheFacet.get())
     repository.attach(componentMaintenanceFacet.get())
-    repository.attach(proxyFacet.get())
+    repository.attach(conanProxyFacet)
     repository.attach(storageFacet.get())
     repository.attach(attributesFacet.get())
     repository.attach(searchFacet.get())
     repository.attach(purgeUnusedFacet.get())
+
   }
 
-  ViewFacet configure(final ConfigurableViewFacet facet) {
+  Closure viewClosure = {ConfigurableViewFacet facet, ConanMatcher matcher ->
```

Comment:
UNRELATED BUT SPREADING STUFF: @jdillon suggested this on some stuff I'm doing for Helm, you might be able to cut down a lot of your Builder stuff doing something similar to what we did here in Helm. The repo is private, I'll link you separate.